Load service configs one by one lazily (#1099)
3scale/APIcast: doc/parameters.md

It can be used instead of `-e` or `---environment` parameter on the CLI and for example
stored in the container image as default environment. Any value passed on the CLI overrides this variable.

> ### `APICAST_LOAD_SERVICES_WHEN_NEEDED`

Author: I wanted to include "lazy" or "lazily" in the name of the ENV. However, I thought that would be a bit confusing because `APICAST_CONFIGURATION_LOADER` already accepts `lazy` as an option. I am open to suggestions for better names :smile:

Reviewer: Some ideas:

1. APICAST_LAZY_LOADING
2. APICAST_LAZY_LOAD_SERVICES
3. APICAST_LAZY_LOAD_CONFIG

@davidor Do you have guidelines or any restrictions when creating env var names?

**Values:**
- `true` or `1` for _true_
- `false`, `0` or empty for _false_

**Default:** _false_

> This option might be useful for users with many services configured. By default,

Reviewer: ```suggestion
This option is useful when there are many services configured. By default,
```

Author: I wrote "might" because it depends a lot on factors such as the number of services, the latency between apicast and the 3scale portal admin, the TTL of the config, etc. We cannot guarantee that this option will always work better than the default.

Reviewer: Got you. This is important.
Suggestion:
`This option can be used when there are many services configured. However, its performance depends on additional factors such as the number of services, the latency between APIcast and the 3scale Admin Portal, the Time To Live (TTL) of the configuration, etc.` <better if you can include a line break before starting the next line with _By default_>

APIcast loads all the services each time it downloads its configuration from the

> admin portal. With a large number of services, this could become problematic.

Reviewer: ```suggestion
Admin Portal. This can cause performance issues with large number of services.
```
